lib/charms/grafana_agent_k8s/v0/grafana_agent.py

```diff
@@ -177,15 +192,53 @@ def _on_promtail_pebble_ready(self, _):
                 "promtail": {
                     "override": "replace",
                     "summary": "promtail",
-                    "command": f"/usr/bin/promtail {self._cli_args}",
-                    "startup": "disable",
+                    "command": f"{WORKLOAD_BINARY_PATH} {self._cli_args}",
```

Comment:
This is not Python 3.5 compliant

Reply:
(I'm gonna review this more later)

I don't think this matters? This charm code will run on Python 3.8, no?

Reply:
True that :-)

Reply:
However for this particular charm, I see a very likely chance that we will want to make a machine subordinate version, hence Python 3.5 overall sounds to me like it makes sense.

Reply:
@Abuelodelanada besides, I thought we said this library would actually be owned by the Loki charm.

Reply:
@mmanciop what I thought we said was the "binary" would be as a Loki resource..... Are you sure you want this in Loki? I thought that our "entrypoint" for logs (and metrics) in LMA will be `grafana-agent`

Reply:
> I thought we said this library would actually be owned by the Loki charm.

Yes, we did that, and I personally still think that's a good idea.

Reply:
Ok, it's not a big deal move the code to Loki.
So, should we provide Loki URL instead of Grafana-agent one to the consumer, right?

Reply:
If you are wondering about the fact that Loki will not use `promtail` ever as a resource, worry you not: how do you think you are going to implement the logging part of self-monitoring for loki-k8s? 😈